src/XrdHttp/XrdHttpUtils.cc

```diff
@@ -496,5 +497,163 @@ char *escapeXML(const char *str) {
   return r;
 }
 
+int mapErrNoToHttp(int errNo) {
 
+  switch (errNo) {
 
+    case EACCES:
+    case EROFS:
+      return HTTP_FORBIDDEN;
+
+    case EPERM:
+      return HTTP_UNAUTHORIZED;
+
+    case ENOENT:
+      return HTTP_NOT_FOUND;
+
+    case EEXIST:
+      return HTTP_CONFLICT;
+
+    case ENOTDIR:
+    case EISDIR:
+    case EXDEV:
+      return HTTP_UNPROCESSABLE_ENTITY;
+
+    case ENAMETOOLONG:
+      return HTTP_URI_TOO_LONG;
+
+    case ELOOP:
+      return HTTP_LOOP_DETECTED;
+
+    case ENOSPC:
+    case EDQUOT:
+      return HTTP_INSUFFICIENT_STORAGE;
+
+    case EFBIG:
+      return HTTP_PAYLOAD_TOO_LARGE;
+
+    case ENOTEMPTY:
+      return HTTP_CONFLICT;
+
+    case EINVAL:
+    case EBADF:
+    case EFAULT:
+    case ENXIO:
+    case ESPIPE:
+    case ENODEV:
+    case EOVERFLOW:
+      return HTTP_BAD_REQUEST;
+
+    case ENOTSUP: // EOPNOTSUPP
+      return HTTP_NOT_IMPLEMENTED;
+
+    case EBUSY:
+    case EAGAIN:
+    case EINTR:
+    case ENOMEM:
+    case EMFILE:
+    case ENFILE:
+    case ETXTBSY:
+      return HTTP_SERVICE_UNAVAILABLE;
+
+    case ETIMEDOUT:
+      return HTTP_GATEWAY_TIMEOUT;
+
+    case ECONNREFUSED:
+    case ECONNRESET:
+    case ENETDOWN:
+    case ENETUNREACH:
+    case EHOSTUNREACH:
+    case EPIPE:
+      return HTTP_BAD_GATEWAY;
+
+    default: //EIO
```

Comment:
That's not just `EIO`, that's all other errors. This comment is misleading, please remove.